streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableReduceTest.java

```diff
@@ -78,4 +92,118 @@ public void shouldAddAndSubtract() {
         strings.addAll(right);
         return strings;
     }
+
+    @Test
+    public void stringSetSerdeTest() {
+        final Set<String> originalSet = new HashSet<>();
+        final String topicName = "serdes";
+
+        final String string1 = "One";
+        final String string2 = "Two";
+        final String string3 = "Three";
+        originalSet.add(string1);
+        originalSet.add(string2);
+        originalSet.add(string3);
+
+        final byte[] bytes = setSerde.serializer().serialize(topicName, originalSet);
+        final Set<String> newSet = setSerde.deserializer().deserialize(topicName, bytes);
+
+        assertTrue(newSet.containsAll(originalSet));
+        assertEquals(originalSet.size(), newSet.size());
+    }
+
+    private static final class StringSetSerde extends WrapperSerde<Set<String>> {
+        public StringSetSerde() {
+            super(new StringSetSerializer(), new StringSetDeserializer());
+        }
+
+        private static class StringSetSerializer implements Serializer<Set<String>> {
+            Serde<String> stringSerde = Serdes.String();
+            Serde<Integer> intSerde = Serdes.Integer();
+
+            @Override
+            public void configure(final Map<String, ?> configs, final boolean isKey) {
+            }
+
+            @Override
+            public byte[] serialize(final String topic, final Set<String> data) {
```

Comment:
Serialization isn't done in the most efficient way, but within the scope of this test is quite sufficient.